sncosmo/__init__.py

```diff
@@ -8,7 +8,23 @@
 from astropy.config import ConfigItem, ConfigNamespace
 from astropy.config.configuration import update_default_config
 
-__version__ = "2.5.0"
+
+# Get the version from setuptools_scm
+try:
+    # Works for Python 3.8+
+    from importlib.metadata import version, PackageNotFoundError
+    __version__ = version("sncosmo")
+except ImportError:
+    # Python 3.7 or lower. Use setuptools to find the version.
+    from pkg_resources import get_distribution, DistributionNotFound
+    try:
+        __version__ = get_distribution("sncosmo").version
+    except DistributionNotFound:
+        # Package is not installed
+        __version__ = '0.0.0'
```

Comment:
Is this the default version we want to use? I ask because I am not fully satisfied with '0.0.0' but I also don't have anything better.

Also, in what environments do expect to get these errors and set `__version__='0.0.0'`?

Reply:
I agree that this isn't a very good default. The environments that you get the errors from are ones that don't have the git history.

Playing around with different use cases a bit more, if you download a tarball of sncosmo from github `setuptools_scm` fails. It also completely prevents you from installing the package. This is also a problem for `astropy`! I'm not a big fan of that, so I'm thinking that reverting to automatically updating the version tag is the way to go.

Reply:
Who downloads tarballs? Do we have a metric on how many people use that?